hypnotoad/core/equilibrium.py

```diff
@@ -3118,6 +3118,22 @@ def plotPotential(
 
         return axis
 
+    def plotWall(self, ax=None):
+        if self.wall:
+            wall_R = [p.R for p in self.wall]
+            wall_Z = [p.Z for p in self.wall]
+
+            # make contours closed
+            wall_R.append(wall_R[0])
+            wall_Z.append(wall_Z[0])
+
+            if ax is None:
+                from matplotlib import pyplot
+
+                pyplot.plot(wall_R, wall_Z, "k-", linewidth=2)
+            else:
+                ax.plot(wall_R, wall_Z, "k-", linewidth=2)
```

Comment:
We can avoid repeating the `plot` call, plus return the new handle:

```suggestion
            if axis is None:
                from matplotlib import pyplot
                fig, axis  = pyplot.subplots()

            axis.plot(wall_R, wall_Z, "k-", linewidth=2)
            return axis
```

Reply:
`pyplot.subplots()` creates a new figure and axes, which isn't what we want. The point of the `axis=None` case is if you're running `hypnotoad` from a Python prompt or a Jupyter notebook and want to do something like
```
plt.figure()
equilibrium.plotPotential()
equilibrium.plotWall()
mesh.plotPoints()
plt.show()
```
I guess we could do something like `from matplotlib import pyplot; axis = pyplot` to avoid the repeated `plot()` call, but I think that's worse than the repeated line.

Agree returning the handle is sensible.

```suggestion
            if axis is None:
                from matplotlib import pyplot

                axis = pyplot.plot(wall_R, wall_Z, "k-", linewidth=2)
            else:
                axis.plot(wall_R, wall_Z, "k-", linewidth=2)

            return axis
```

Reply:
Ah, I think we have differing views on how we expect this to work :) I'm not a big fan of the matlab-style state machine approach, so I would expect three new figures with individual plots from your example. Passing `axis=None` is a convenience for creating a new figure/axis. Your example is easily adapted to plot on a single axis by passing `axis=gca()`

Reply:
I get the reservations with the state-machine thing. It's unlikely to be a big deal, because especially now we have the gui it's not likely users will be running from a Python prompt. I do still think that users will expect `pyplot`-style behaviour - in particular `xarray`'s plot functionality behaves that way (although the ones in `xBOUT`'s `bout` accessor don't because those plot methods don't really make sense to layer onto an existing plot).

Practically, I don't see a strong use-case in `hypnotoad` for "`axis=None` creates a new figure", but I do see potential convenience in allowing `pyplot`-style updates to an existing plot, so I'm inclined to keep that.